pytest_rts/tests/test_e2e.py

```diff
@@ -1,215 +1,32 @@
-
-    # Run tool -> db updated to commit1 state
-    helper.run_tool()
-
-    helper.change_file("changes/shop/change_get_price.txt", "src/shop.py")
-    helper.commit_change("src/shop.py", "commit2")
-
-    helper.squash_commits(3, "commit3")
-
-    # Tool should find only tests for change introduced in commit1 -> commit2
-    assert helper.get_tests_from_tool_committed() == {
-        "tests/test_shop.py::test_normal_shop_purchase",
-        "tests/test_shop.py::test_normal_shop_purchase2",
-    }
-
-
-def test_init_code_tracked(helper):
-    helper.change_file("changes/init/change_function_one.txt", "src/__init__.py")
-    assert helper.get_tests_from_tool_current() == {"tests/test_init.py::test_one"}
-
-
-def test_decorated_tracked(helper):
-    """Test that changes in the decorator switch on the test using decorated
-    function.
+def test_only_new_functions_are_ran(testdir):
+    """Test case for running pytest-rts when new tests are added
+    and changes are not committed.
     """
-    helper.change_file("changes/decorated/change_decorator.txt", "src/decorators.py")
-    assert helper.get_tests_from_tool_current() == {
-        "tests/test_decorated.py::test_decorated"
-    }
+    # copy the helper project
+    testdir.copy_example(".")
 
+    # run pytest pytest-cov to produce mapping
+    # subprocess used to not mess up with coverage for this test suite
+    testdir.runpytest_subprocess("--cov=.", "--cov-context=test")
 
-def test_decorated_with_param_tracked(helper):
-    """Test that changes in the parametrized decorator switch on the test using
-    decorated function.
-    """
-    helper.change_file("changes/decorated/change_decorator_2.txt", "src/decorators.py")
-    assert helper.get_tests_from_tool_current() == {
-        "tests/test_decorated.py::test_decorated_2"
-    }
+    # rename .coverage file produced by pytest-cov
+    os.rename(".coverage", "rts-coverage.1234")
 
+    # add a new test method to an existing file
+    os.rename("changes/test_car/add_test_passengers.txt", "tests/test_car.py")
 
-def test_testfunction_runtimes_not_wiped(helper):
-    """Test that checks that test function runtimes are not removed
-    from database when running the tool for committed changes
-    and a testfile is changed
+    # add a new test file with a test in it
+    testdir.makepyfile(
+        """
+        def test_empty():
+            assert 1 == 1
     """
-    orig_runtimes = helper.get_test_function_runtimes()
-    helper.checkout_new_branch()
-
-    helper.change_file("changes/test_shop/shift_two_forward.txt", "tests/test_shop.py")
-    helper.commit_change("tests/test_shop.py", "shift")
-
-    helper.run_tool()
-    new_runtimes = helper.get_test_function_runtimes()
+    )
 
-    assert orig_runtimes == new_runtimes
+    # run pytest-rts to only run new tests
+    result = testdir.runpytest("--rts", "--rts-coverage-db=rts-coverage.1234")
+    result.assert_outcomes(passed=2, failed=0, errors=0)
```

Comment:
not critical: if it is easy to check test names then it would be a better assertion

Reply:
Unfortunately there doesn't seem to be an easy option to do this